sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -86,10 +89,28 @@ case class HadoopFsRelation(
   }
 
   override def sizeInBytes: Long = {
```

Comment:
May be you need to implement a rule similar to `DetermineTableStats` for the datasource table?

Reply:
Hi @wangyum , The issue here  is catalogFileIndex always take stats as default stats and it never gets updated, even if the user enable 'fallBackToHdfsForStats'
So, In this fix, if the user enable the 'fallBackToHdfsForStats', it reads the sizeInBytes from the fileSystem, rather than relying on the default table stats.
Thanks